src/Microsoft.Health.Dicom.Metadata/Registration/DicomServerBuilderMetadataRegistrationExtensions.cs

```diff
@@ -34,33 +31,23 @@ public static IDicomServerBuilder AddMetadataStorageDataStore(this IDicomServerB
             EnsureArg.IsNotNull(configuration, nameof(configuration));
 
             return serverBuilder
-                        .AddMetadataPersistence(configuration)
-                        .AddMetadataHealthCheck();
+                .AddMetadataPersistence(configuration)
+                .AddMetadataHealthCheck();
         }
 
         private static IDicomServerBuilder AddMetadataPersistence(this IDicomServerBuilder serverBuilder, IConfiguration configuration)
         {
             IServiceCollection services = serverBuilder.Services;
 
-            services.AddBlobDataStore();
```

Comment:
I assume that most of what was in `AddBlobDataStore` is generally now considered to be called directly. Should we mark it as deprecated or provide guidance in the documentation for it?

Reply:
It differs a bit from `AddBlobDataStore` in that the previous method combined:
1. Registration of a `BlobServiceClient`
2. Registration of `BlobDataStoreConfiguration`
3. Registration of blob initialization hosted service.

Whereas today I've split them out so provide some more idiomatic APIs and customization options. I've also tried to remove the dependency on `BlobDataStoreConfiguration` and instead use configurations that `BlobServiceClient` can directly interface with. That way we don't need to map every setting ourselves. You're right though. I should probably move to deprecate the old API if everything works out fine with this for us in DICOM.